drools-impact-analysis-graph/drools-impact-analysis-graph-common/src/main/java/org/drools/impact/analysis/graph/ModelToGraphConverter.java

```diff
@@ -0,0 +1,221 @@
+        // all rules which react to the fact
+        Set<Rule> reactedRules = reactFieldMap.values().stream().flatMap(ruleSet -> ruleSet.stream()).collect(Collectors.toSet());
+        Node source = nodeMap.get(fqdn(pkgName, ruleName));
+        for (Rule reactedRule : reactedRules) {
+            Node target = nodeMap.get(fqdn(pkgName, reactedRule.getName()));
+            Node.linkNodes(source, target, Link.Type.POSITIVE);
+        }
+    }
+
+    private void processDelete(Map<String, Node> nodeMap, Map<Class<?>, Map<String, Set<Rule>>> reactFactMap, String pkgName, String ruleName, ConsequenceAction action) {
+        // TODO: consider exists()
+        Class<?> deletedClass = action.getActionClass();
+        Map<String, Set<Rule>> reactFieldMap = reactFactMap.get(deletedClass);
+        // all rules which react to the fact
+        Set<Rule> reactedRules = reactFieldMap.values().stream().flatMap(ruleSet -> ruleSet.stream()).collect(Collectors.toSet());
+        Node source = nodeMap.get(fqdn(pkgName, ruleName));
+        for (Rule reactedRule : reactedRules) {
+            Node target = nodeMap.get(fqdn(pkgName, reactedRule.getName()));
+            Node.linkNodes(source, target, Link.Type.NEGATIVE);
+        }
+    }
+
+    private void processModify(Map<String, Node> nodeMap, Map<Class<?>, Map<String, Set<Rule>>> reactFactMap, String pkgName, String ruleName, ModifyAction action) {
+        // TODO: consider exists()/not()
+        Node source = nodeMap.get(fqdn(pkgName, ruleName));
+
+        Class<?> modifiedClass = action.getActionClass();
+        Map<String, Set<Rule>> reactFieldMap = reactFactMap.get(modifiedClass);
+        List<ModifiedProperty> modifiedProperties = action.getModifiedProperties();
+        for (ModifiedProperty modifiedProperty : modifiedProperties) {
+            String property = modifiedProperty.getProperty();
+            Set<Rule> rules = reactFieldMap.get(property);
+            for (Rule reactedRule : rules) {
+                List<Constraint> constraints = reactedRule.getLhs().getPatterns().stream()
+                                                          .filter(pattern -> pattern.getPatternClass() == modifiedClass)
+                                                          .flatMap(pattern -> pattern.getConstraints().stream())
+                                                          .filter(constraint -> constraint.getProperty().equals(property))
+                                                          .collect(Collectors.toList());
+                if (constraints.size() == 0) {
+                    // This rule is reactive to the property but cannot find its constraint. Exception or UNKNOWN?
+                    throw new RuntimeException("This rule [" + reactedRule.getName() + "] is reactive to the property [" + property + "] " +
+                                               "but cannot find its constraint : \n" + reactedRule);
+                }
```

Comment:
Currently non-literal constraint (e.g. age > $a) is not parsed to Constraint, right? In this case, I throw a RuntimeException for now. But I may be able to take it as an "UNKNOWN" link.

Reply:
@tkobayas In reality this is not a missing part yet to be implemented, but something wanted and expected. I don't see (but correct me if I'm wrong) how having a constraint saying that you have the predicate `age > $a` could be useful to build the impact analysis graph. If you have a comparison like `age > 18` you can check if there's a rule doing a `modify` with a literal value greater than 18, but what you can do when it is not a literal? In that case the only information that you need is that there's a constraint on the field `age` but you already have it in the property reactivity list of the pattern where this constraint is defined.

Reply:
@mariofusco Thanks, that makes sense. So if Rule1 modifies a property "age" and Rule2 has "age" in reactOnFields but doesn't have Constraint of "age", it means that Rule1 impacts Rule2 with UNKNOWN (not POSITIVE/NEGATIVE) way. I will update accordingly.